lib/pbxProject.js

```diff
@@ -1309,15 +1330,16 @@ pbxProject.prototype.addToOtherLinkerFlags = function (flag) {
     }
 }
 
-pbxProject.prototype.removeFromOtherLinkerFlags = function (flag) {
+pbxProject.prototype.removeFromOtherLinkerFlags = function (flag, opt) {
     var configurations = nonComments(this.pbxXCBuildConfigurationSection()),
         OTHER_LDFLAGS = 'OTHER_LDFLAGS',
         config, buildSettings;
 
     for (config in configurations) {
         buildSettings = configurations[config].buildSettings;
 
-        if (unquote(buildSettings['PRODUCT_NAME']) != this.productName) {
+        var productName = opt && opt.productName ? opt.productName : this.productName;
+        if (unquote(buildSettings['PRODUCT_NAME']) != productName) {
```

Comment:
please use strict equal checks

```suggestion
        if (unquote(buildSettings['PRODUCT_NAME']) !== productName) {
```